pkg/ioutils/readers.go

```diff
@@ -189,6 +190,9 @@ func (r *bufReader) drain() {
 		reuseCount++
 		r.wait.Signal()
 		r.Unlock()
+		//allow or force Go scheduler to switch context, without explicitly
+		//forcing this will make it hang when using gccgo implementation
+		runtime.Gosched()
```

Comment:
we already have build flags in other parts of the code for gccgo, readability is not an issue. You can extract that call into a function, with the `+build gccgo` flag and then have another function that does nothing for `+build !gccgo`.